### PR TITLE
Backend integration (fetch common miscLabels and all schools/courses)

### PR DESCRIPTION
**Pull request by andres-holguin**
This PR integrates the changes on the backend made by @Aradhya-B in #57 and #60 into the FrontEnd
### Changes
- The UploadForm will now fetch the school and courses in the Database to display as suggestions to the user
- The UploadForm will now fetch the most common labels from the Database to display as suggestions to the user
- The FilterList will now fetch the most common labels from the Database to display as suggestions for filtering to the user.
- The content in index.html has been moved to a Vue component under the views/ folder. This is initiates a project structure more in line with modern webapps.
- Various TODOs were resolved and placeholder content was eliminated.
-  The test data from MockSearchButton.vue was enhanced.

### CODE REVIEW

File: src/main/webapp/views/SearchPage.vue [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        a way of counting the # of favorites and downloads a share has had in the past X days.
+      -->
+      <!--
+      <note-grid-collection :note-data="searchResult"
+                            header="Trending Notes"
+                            :filters="filters"
+                            :compare-func="descFavorites"
+                            :max-age="7 * 24 * 60 * 60 * 1000">
+      </note-grid-collection>
+      -->
```
Comment (Aradhya-B) [inline src/main/webapp/views/SearchPage.vue:29]: This is interesting, but I'm not 100% sure how we'd go about implementing a servlet to get "trending" notes on the backend with the existing schema. I feel like we'd need a `date_favorited` field on the favorite_notes table and to track downloads, we'd need a new downloaded_notes table that tracks a `note_id`, `user_id`, and `date_downloaded`. Having these tables also opens the possibility in the future for a note recommendations feature that sees all the labels of the notes a user has downloaded and favorited and shows them notes that have the same or similar labels.

**Review by Aradhya-B — APPROVED** [review]
LGTM :), great work!!

File: src/main/webapp/views/SearchPage.vue (hunk 1 above)
Comment (andres-holguin) [reply src/main/webapp/views/SearchPage.vue:29]: ^ That's a cool idea for a new feature.

**Review by keelywan — APPROVED** [review]
Looks good!

**Review by nickKelman — APPROVED** [review]
Nice work

**Pull request merged**